lib/active_merchant/billing/gateways/latitude19.rb

```diff
@@ -0,0 +1,411 @@
+          auth_or_sale("sale", nil, amount, payment_method, options)
+        else
+          MultiResponse.run() do |r|
+            r.process { get_session(options) }
+            r.process { get_token(r.authorization, payment_method, options) }
+            r.process { auth_or_sale("sale", r.authorization, amount, payment_method, options) }
+          end
+        end
+      end
+
+      def authorize(amount, payment_method, options={})
+        if options[:account_token]
+          auth_or_sale("auth", nil, amount, payment_method, options)
+        else
+          MultiResponse.run() do |r|
+            r.process { get_session(options) }
+            r.process { get_token(r.authorization, payment_method, options) }
+            r.process { auth_or_sale("auth", r.authorization, amount, payment_method, options) }
+          end
+        end
+      end
+
+      def capture(amount, authorization, options={})
+        post = {}
+        post[:method] = "deposit"
+        add_request_id(post)
+
+        params = {}
+
+        _, params[:pgwTID] = split_authorization(authorization)
+
+        add_invoice(params, amount, options)
+        add_credentials(params, post[:method])
+
+        post[:params] = [params]
+        commit("v1/", post)
+      end
+
+      def void(authorization, options={})
+        method, pgwTID = split_authorization(authorization)
+        case method
+        when "auth"
+          reverse_or_void("reversal", pgwTID, options)
+        when "deposit", "sale"
+          reverse_or_void("void", pgwTID, options)
+        else
+          message = "Unsupported operation. Only successful and active - Purchase, Authorize and Capture transactions can be voided."
+          return Response.new(false, message)
+        end
+      end
+
+      def credit(amount, payment_method, options={})
+        if options[:account_token]
+          refundWithCard(nil, amount, payment_method, options)
+        else
+          MultiResponse.run() do |r|
+            r.process { get_session(options) }
+            r.process { get_token(r.authorization, payment_method, options) }
+            r.process { refundWithCard(r.authorization, amount, payment_method, options) }
+          end
+        end
+      end
+
+      def verify(payment_method, options={}, action=nil)
+        if options[:account_token]
+          verifyOnly(action, nil, payment_method, options)
+        else
+          MultiResponse.run() do |r|
+            r.process { get_session(options) }
+            r.process { get_token(r.authorization, payment_method, options) }
+            r.process { verifyOnly(action, r.authorization, payment_method, options) }
+          end
+        end
+      end
+
+      def store(payment_method, options={})
+        verify(payment_method, options, "store")
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((\"pgwAccountNumber\\\":\\\")\d+), '\1[FILTERED]').
```

Comment:
@rwdaigle Alright.

hmac goes out with every request type except one, and the one without hmac contains PAN that's going to get scrubbed. so, it should be good. leaving tokens un-scrubbed  :+1: